bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/ThingType.java

```diff
@@ -286,6 +302,10 @@ public String getBindingId() {
                     }
                 }
             }
+
+            if (ref != null) {
+                bundleContext.ungetService(ref);
```

Comment:
Shouldn't we use a final block to ensure the service reference is released?

Reply:
Yes